pkg/arvo/ted/ddiff.hoon

```diff
@@ -0,0 +1,297 @@
+/-  spider
+/+  strandio
+=,  strand=strand:spider
+=,  clay
+^-  thread:spider
+|=  arg=vase
+=/  m  (strand ,vase)
+^-  form:m
+|^
+=+  !<([~ =a=path =b=path shallow=flag] arg)
+=/  a-beam  (need (de-beam a-path))
+=/  b-beam  (need (de-beam b-path))
+;<  a-dome=dome  bind:m  (get-from-clay a-beam dome %v)
+;<  b-dome=dome  bind:m  (get-from-clay b-beam dome %v)
```

Comment:
It looks like the ankh doesn't appear when you fetch the dome from a past revision or a foreign desk.  The ankh is fundamentally a cache, and we only maintain it for the latest revision of local desks.

In those cases, I think the right move is to request the %v like you do here so that we download the desk if it's foreign, and then use `%y` (which produces `arch`, which is `[fil=(unit lobe) dir=(map @ta ~)]`) requests instead of inspecting the ankh.  Since this will work for all cases, it's probably best to do everything that way.